crates/telemetry/src/state.rs

```diff
@@ -1168,3 +1200,116 @@ impl AggregatorSelector for CustomAggregatorSelector {
         }
     }
 }
+
+#[cfg(feature = "telemetry")]
+mod tests {
```

Comment:
This could probably use a `cfg(test)` attribute as well:

```suggestion
#[cfg(test)]
mod tests {
```